doc/PORTABLE_SERVICES.md

```diff
@@ -0,0 +1,119 @@
+entirely generic, and doesn't have to carry any specific metadata beyond what
+distribution images carry anyway.
+
+If you so will, "Portable Services" are a nicer way to manage chroot()
+environments, with better security, tooling and behaviour.
+
+# Where's the different to a "Container"?
+
+"Container" is a very vague term, after all it is used for
+systemd-nspawn/LXC-type OS containers, for Docker/rkt-like micro service
+containers, and even certain 'lightweight' VM runtimes.
+
+The "portable service" concept ultimately will not provide a fully detached
+environment to the payload, like containers intend to. Instead they are
+primarily much like regular system services, can be controlled with the same
+tools, are exposed the same way in all tools and so on. Their main difference
+is that the use a different root directory than the rest of the system. Hence,
+the intention is not to run code in a different, isolated world from the host —
+like most containers would do it —, but to run it in the same world, but with
+stricter access controls on what the service can see and do.
+
+As one point of differentiation: as programs run as "portable services" are
+pretty much regular system services, they won't run as PID 1 (like Docker would
+do it), but as normal process. A corrolary of that is that they aren't supposed
+to manage anything in their own system, such as the network, which is shared
+with the rest of the system.
+
+The primary focus use case of "portable services" is to extend the host system
+with encapsulated extensions, but provide almost full integration with the rest
+of the system, though possibly restricted by effective security knobs. This
+focus includes system extensions otherwise often called "super-privileged
+containers".
+
+Note that portable services are only available for system services, not for
+user services. i.e. the functionality cannot be used for the stuff
+bubblewrap/flatpak is focussing on.
+
+# Mode of Operation
+
+If you have portable service image, maybe in a raw disk image called
+`foobar.raw`, then attaching the services to the host is as easy as:
+
+```
+# /usr/lib/systemd/portablectl attach foobar.raw
+```
+
+This command does the following:
+
+1. It dissects the image, checks and validates the `/etc/os-release` data of
+   the image, and looks for all included unit files.
+
+2. It copies out all unit files with a suffix of `.service`, `.socket`,
+   `.target` and `.path` whose name begins with the image's name (with the .raw
+   removed), truncated at the first underscore (if there is one). This prefix
+   name generated from the image name must be followed by a "." or "-"
+   character in the unit name. Or in other words, given the image name of
+   `foobar.raw` all unit files matching `foobar-*.{service|socket|target|path}`
+   as well as `foobar.*.{service|socket|target|path}` and
+   `foobar.{service|socket|target|path}` are copied out. These unit files are
```

Comment:
From anywhere inside the image or just the root directory? Will it traverse the directory tree looking for any files with those names? Wasn't clear from my reading...

Reply:
it looks at the typical unit file directories within the image. i.e. the stuff below /etc/systemd/system/ as well as /usr/lib/systemd/system, and so on... The idea is really that the image can be as regular an OS image as it can be, and we just reuse the stuff that is already there and right from the places it would be in a regular OS image